kubernetes-client/test/src/com/goyeau/kubernetes/client/cache/AuthorizationCacheTest.scala

```diff
@@ -73,6 +74,21 @@ class AuthorizationCacheTest extends FunSuite {
     io.unsafeRunSync()
   }
 
+  test(s"retrieve the token when it's going to expire within refreshBeforeExpiration") {
+    val io = for {
+      counter <- IO.ref(1)
+      auth = mkAuthorization(
+        expirationTimestamp = IO.realTimeInstant.map(_.plusSeconds(8).some),
+        token = counter.getAndUpdate(_ + 1).map(i => s"test-token-$i")
+      )
+      cache    <- AuthorizationCache[IO](retrieve = auth, refreshBeforeExpiration = 10.seconds)
```

Comment:
might be worth adding a "few" seconds to the `refreshBeforeExpiration` - in case the test takes more than 2 seconds to run for whatever reason